### PR TITLE
added detail report screen

### PR DESCRIPTION
**Pull request by tonae-patterson**
Complete. 
I changed the name of the date column to info because it includes entry notes in the toggle. If date would be better the text can be altered.

### CODE REVIEW

**Comment by jwaltt55** [comment]
Instead of having each time entry listed on this page be a drop-down menu, I think it'd be better to just have their comments listed out by default. If someone wants to see the comments for each time entry in the list, they'd have to manually scroll down and expand every single time entry the way we have it rn. Another idea could be to just have an "expand all" option or something that could expand/de-expand all entries at once.

**Comment by jwaltt55** [comment]
I think it'd also be a little nicer to have the check-boxes for the team members to be outside of a drop-down menu as well, just so that you don't have to click on that menu every time to see who you currently have selected.

**Comment by jwaltt55** [comment]
It might also be nicer if the time entries list wasn't a fixed height with it's own scrollbar, just so that more entries could be shown on the screen at once.

**Comment by dyl10s** [comment]
Keep in mind that prof Adams wanted to be able to see the last modified time as well.

**Comment by dyl10s** [comment]
I also noticed when changing the members selection an API call is made. You don't need to make this call because each time you request the API all of the information for each user is returned, you could just store the results in a variable or something, it will make the app much quicker. You are making a request on the time frame changes with is correct 👍

**Comment by tonae-patterson** [comment]
Everything has been implemented, the only issue remaining is making the toggle row display in mobile only:

- Cant get nebulars (HideOn / ShowOn) tree grid column feature to work at all.  The column remains in view. I think it can only be implemented using nebulars NbTreeGridColumnDefDirective somehow, but I have been trying things all night.
- This also brings to my attention that there exists a child row either way, so I'm unsure if it is possible to hide children in desktop without implementing a separate data source. Although the toggle arrow is hidden in desktop, you can still click the rows and toggle.

**Comment by jwaltt55** [comment]
It looks like even though the drop-down arrows are gone and the comments are listed in their own column now which looks good, you can still click on the table rows and have them expand into an extra empty row.

**Comment by jwaltt55** [comment]
This is also a pretty small thing but we probably want the "Data Ascending/Descending" button to not have sharp corners on its right side unless it's right next to another button.

**Comment by jwaltt55** [comment]
Also, I still personally think it'd look a little better if the time entries table just went down the page instead of having a fixed-height with its own scrollbar, unless we really do wanna keep it that way.

**Comment by jwaltt55** [comment]
I think it might be better if the "Created" column was just named "Date" or something, because the value it's displaying is the date the time entry is for, not the date it was created on.

**Comment by jwaltt55** [comment]
I like the sorting by clicking on a column feature, but it might be good to somehow make it more clear to the user that they can do that, maybe by having the little arrow already displayed when the page loads or highlighting a clickable column when you hover over it.

**Comment by jwaltt55** [comment]
Also, the responsive behavior seems a little weird, it looks like time range drop-down menu expands before the member drop-down menu does, and if you make the screen narrow enough the table starts to get cut off as well.

**Comment by tonae-patterson** [comment]
> Also, the responsive behavior seems a little weird, it looks like time range drop-down menu expands before the member drop-down menu does, and if you make the screen narrow enough the table starts to get cut off as well.

@jwaltt55 I implemented your other feedback, but I haven't noticed a difference in responsiveness of the dropdown menus. If there was any delay it could be due to the range dropdown menu having pre-defined values, whereas the member dropdown is defined and loaded dynamically.

**Comment by dyl10s** [comment]
I think this is what Justin was talking about with the date range drop-down expanding before the member dropdown, 
image

Also it looks like the member dropdown is not as thick as the date-range drop down, probably want to just make them the same. I feel like it may be better just to show a simplified report on mobile, maybe just date, member and hours. I feel like a scrollable grid is kinda hard to follow.

File: Frontend/src/app/views/dashboard/report-details/report-details.component.scss [hunk 1]
```diff
@@ -58,14 +60,9 @@
...
+}
 
-  .filter-result-section {
-    table {
-      display: table;
-      overflow: hidden;
-    }
-  }
-
+@media only screen and (min-width: 830px) {
```
Comment (dyl10s) [inline Frontend/src/app/views/dashboard/report-details/report-details.component.scss]: Change this to 960px as it is when the date range picker goes to mobile view

**Review by dyl10s — CHANGES_REQUESTED** [review]
Just this one last thing and then I think we are good to go

**Review by dyl10s — APPROVED** [review]
Looks good!
